### ISSUE TITLE
MethodOverride improvements

### ISSUE THREAD

**Issue by designermonkey** [oscarotero/psr7-middlewares#28]
In my adventures, I tend to find that the use case for overriding an HTTP method comes from web browsers where we all know there is poor support for anything other than POST and GET.

I'm just wondering how useful this middleware package is when solely relying on a header to do the overriding? Generally, clients that can set headers can also make requests with deliberate HTTP methods.

Would it be possible to have this upgraded to use a POST form field, and query-string parameter alongside the header?

``` php
$getUrl = "example.com?http-method-override=HEAD";

$postBody = [
    "http-method-override" => "PUT"
];
```

It would become so much more useful in the wild with additions like this.

**Comment by oscarotero** [comment]
I like this idea. After a quick search I found other similar middlewares providing this feature, some only use the post body, others also the query string parameteres. What do you think about something like this?:

``` php
Middleware::methodOverride()
    ->parameter('http-method-override') //To use this parameter in the parsed body and url query 
    ->parameter('http-method-override', false) //To use only in the parsed body
```

**Comment by designermonkey** [comment]
Yeah, that sounds great!

**Comment by oscarotero** [comment]
Released 3.14 with this feature :)

**Comment by designermonkey** [comment]
Brilliant! Cheers.
